ClemBot.Bot/bot/cogs/search_cog.py

```diff
@@ -0,0 +1,159 @@
+related_topics_pattern = re.compile('(https://duckduckgo.com/)([acdne]/)?(\\S+)', re.IGNORECASE)
+
+
+class Category(Enum):
+    ARTICLE = 'A', 'Article'
+    DISAMBIGUATION = 'D', 'Disambiguation'
+    CATEGORY = 'C', 'Category'
+    NAME = 'N', 'Name'
+    EXCLUSIVE = 'E', 'Exclusive'
+    NONE = 'None', 'None'
+
+    def code(self):
+        return self.value[0]
+
+    def name(self):
+        return self.value[1]
+
+
+class SearchResult:
+
+    def __init__(self, json_response):
+        self.json_response = json_response
+
+    def __getitem__(self, item):
+        return self.json_response
+
+    def has_result(self):
+        return len(self.json_response['Heading']) > 0
+
+    def title(self):
+        return self.json_response['Heading']
+
+    def abstract(self):
+        return markdownify(self.json_response['Abstract'])
+
+    def category(self):
+        return category_from_code(self.json_response['Type'])
+
+    def has_thumbnail(self):
+        return len(self.json_response['Image']) > 0
+
+    def thumbnail(self):
+        # surprisingly 'Image' in the JSON returns the latter half of the URL,
+        # so we need to concat the other half, i.e., https://duckduckgo.com
+        return image_url + self.json_response['Image']
+
+    def related_topics(self) -> list:
+        # related topics can return topics that we don't actually want,
+        # so we're taking the subset of data that we can use and returning it
+        topics = list()
+        related_topics = self.json_response['RelatedTopics']
+        for i in range(0, len(related_topics)):
+            topic = related_topics[i]
+            if 'FirstURL' not in topic:
+                break
+            topics.append(topic)
+        return topics
+
+    def has_related_topics(self):
+        return len(self.related_topics()) > 0
+
+    def related_topics_formatted(self):
+        related_topics = self.related_topics()
+        related_topics_formatted = list[str]()
+        # maximum related topics can be changed in the global vars
+        for i in range(0, min(max_related_topics, len(related_topics))):
+            topic = related_topics[i]
+            # stupid way of getting short related topics titles
+            title = related_topics_pattern\
+                .match(topic['FirstURL'])\
+                .group(3)\
+                .replace('_', ' ')
+            title = unquote_plus(title)
+            related_topics_formatted.append(f"[{title}]({topic['FirstURL']})")
+        return '\n'.join(related_topics_formatted)
+
+    def url(self):
+        return self.json_response['AbstractURL']
+
+    def source(self):
+        return self.json_response['AbstractSource']
+
+
+class SearchCog(commands.Cog):
+
+    def __init__(self, bot):
+        self.bot = bot
+
+    @ext.command()
+    @ext.short_help('Searches via DuckDuckGo.')
+    @ext.long_help('Searches via DuckDuckGo\'s instant web API to bring up results from Wikipedia, wikiHow, and more.')
```

Comment:
For strings the want to use a single quote ' in the body we can use double " on the outside